builder/azure/arm/step_get_source_image_name.go

```diff
@@ -36,29 +34,9 @@ func (s *StepGetSourceImageName) Run(ctx context.Context, state multistep.StateB
 	}
 
 	if s.config.SharedGallery.Subscription != "" {
-		client := s.client.GalleryImageVersionsClient
-		client.SubscriptionID = s.config.SharedGallery.Subscription
-
-		image, err := client.Get(ctx, s.config.SharedGallery.ResourceGroup,
-			s.config.SharedGallery.GalleryName, s.config.SharedGallery.ImageName, s.config.SharedGallery.ImageVersion, "")
-
-		if err != nil {
-			log.Println("[TRACE] unable to derive managed image URL for shared gallery version image")
-			s.GeneratedData.Put("SourceImageName", "ERR_SOURCE_IMAGE_NAME_NOT_FOUND")
-			return multistep.ActionContinue
-		}
-
-		if image.GalleryImageVersionProperties != nil && image.GalleryImageVersionProperties.StorageProfile != nil &&
```

Comment:
When we wrote this logic you couldn't directly publish a shared image gallery image version from Packer, now you can, when you set that HCP Packer doesn't expect the Source ID of the parent Gallery Image Version (this would be like, the specific Windows marketplate image), for the parent, it expects the gallery image itself, because of that we can just infer the SIG resource ID and don't need to make the extra request